src/rgw/driver/rados/rgw_rados.cc

```diff
@@ -6532,18 +6532,27 @@ int RGWRados::set_attrs(const DoutPrefixProvider *dpp, RGWObjectCtx* octx, RGWBu
       bufferlist acl_bl;
       if (iter = attrs.find(RGW_ATTR_ACL); iter != attrs.end()) {
         acl_bl = iter->second;
+      } else if (auto i = state->attrset.find(RGW_ATTR_ACL);
+                 i != state->attrset.end()) {
+        acl_bl = i->second;
       }
       std::string etag;
       if (iter = attrs.find(RGW_ATTR_ETAG); iter != attrs.end()) {
         etag = rgw_bl_str(iter->second);
+      } else if (state->attrset.find(RGW_ATTR_ETAG) != state->attrset.end()) {
+        etag = rgw_bl_str(state->attrset.find(RGW_ATTR_ETAG)->second);
```

Comment:
nit: these should store the iterator instead of calling `find()` twice, like you did above for RGW_ATTR_ACL